src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,124 @@
+        for (line in lore) {
+            neededPetPattern.matchMatcher(line) {
+                val origTierString = group("tier")
+                val origPetString = group("pet")
+                val pet = origPetString.uppercase().replace(" ", "_").removePrefix("FROST_")
+                val originalTier = LorenzRarity.getByName(origTierString.uppercase().replace(" ", "_"))?.id ?: 0
+                val (cheapestTier, petPrice) = findCheapestTier(pet, originalTier)
+                val displayPetString =
+                    if (cheapestTier == originalTier) group("fullThing")
+                    else "${LorenzRarity.getById(cheapestTier)?.formattedName} $origPetString"
+                if (petPrice != -1.0) {
+                    totalCost += petPrice
+                    this@buildList.add(
+                        Renderable.clickAndHover(
+                            text = " §7- $displayPetString§7: §6${petPrice.addSeparators()} coins",
+                            tips = listOf(
+                                "§aClick to run §e/ahs ] $origPetString §ato find it on the Auction House.",
+                                "§aNotes: §eSet the rarity filter yourself. §cBooster Cookie required!"
+                            ),
+                            onClick = { LorenzUtils.sendCommandToServer("ahs ] $origPetString") }
+                        ))
+                } else {
+                    this@buildList.add(
+                        Renderable.clickAndHover(
+                            text = " §7- $displayPetString§7: §eNot on AH; view its Wiki article here.",
+                            tips = listOf("§4Click to run §e/wiki $pet §4to view how to obtain it."),
+                            onClick = { LorenzUtils.sendCommandToServer("wiki $pet") }
+                        ))
+                }
+            }
+        }
+        this.add(Renderable.string("§dTotal cost §7(§6Lowest BIN§7): §6${totalCost.addSeparators()} coins"))
+        if (config.otherRarities) this.add(Renderable.string("§c§lDisclaimer:§r§c Total does not include costs to upgrade via Kat."))
+    }
+
+    private fun findCheapestTier(pet: String, originalTier: Int): Pair<Int, Double> {
+        val petPriceOne = petInternalName(pet, originalTier).getPetPrice()
+        val petPrices= mutableListOf(petPriceOne)
+        if (config.otherRarities || petPriceOne == -1.0) {
+            petPrices.add(petInternalName(pet, originalTier - 1).getPetPrice(otherRarity = true))
+            if (originalTier != 5) petPrices.add(petInternalName(pet, originalTier + 1).getPetPrice(otherRarity = true))
+        }
+        val tierAndIndex = petPrices.withIndex().minBy { it.value }
+        return Pair(tierAndIndex.cheapestTierIndex(originalTier), tierAndIndex.value)
+    }
+
+    private fun IndexedValue<Double>.cheapestTierIndex(originalTier: Int) = when (this.index) {
+        1 -> originalTier - 1
+        2 -> originalTier + 1
+        else -> originalTier
+    }
```

Comment:
This I think is the best approach. Ery, this is no absolute, but this is how I would do it.
```suggestion
    private fun findCheapestTier(pet: String, originalTier: Int) = buildMap<Int, Double> {
        this[originalTier] = petInternalName(pet, originalTier).getPetPrice()
        if (config.otherRarities || this[originalTier] == -1.0) {
            this[originalTier - 1] = petInternalName(pet, originalTier - 1).getPetPrice(otherRarity = true)
            if (originalTier != 5) {
                this[originalTier + 1] = petInternalName(pet, originalTier + 1).getPetPrice(otherRarity = true)
            }
        }
    }.minBy { it.value }
```